spec.html

```diff
@@ -56,21 +25,41 @@ <h1>Array.prototype.popped ( )</h1>
 
                 <emu-alg>
                     1. Let _O_ be ? ToObject(*this* value).
-                    1. Let _A_ be ArrayClone(_O_).
-                    1. Perform ? Call(Array.prototype.pop, _A_, «»).
+                    1. Let _len_ be ? LengthOfArrayLike(_O_).
+                    1. Let _newLen_ be max(_len_ - 1, 0).
+                    1. Let _A_ be ? ArraySpeciesCreate(_O_, 𝔽(_newLen_)).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _newLen_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _elementK_ be ? Get(_O_, _Pk_).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _elementK_).
+                        1. Set _k_ to _k_ + 1.
                     1. Return _A_.
                 </emu-alg>
             </emu-clause>
 
             <emu-clause id="sec-array.prototype.pushed">
-                <h1>Array.prototype.pushed ( ..._args_ )</h1>
+                <h1>Array.prototype.pushed ( ..._items_ )</h1>
 
                 <p>When the *pushed* method is called, the following steps are taken:</p>
 
                 <emu-alg>
                     1. Let _O_ be ? ToObject(*this* value).
-                    1. Let _A_ be ArrayClone(_O_).
-                    1. Perform ? Call(Array.prototype.push, _A_, «..._args_»).
+                    1. Let _itemCount_ be the number of elements in _items_.
+                    1. Let _len_ be ? LengthOfArrayLike(_O_).
+                    1. let _newLen_ be _len_ + _itemCount_.
+                    1. Let _A_ be ? ArraySpeciesCreate(_O_, 𝔽(_newLen_)).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _len_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _elementK_ be ? Get(_O_, _Pk_).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _elementK_).
+                        1. Set _k_ to _k_ + 1.

```

Comment:
these steps still seem duplicated both here and in popped - could this be an AO that takes `O` and `newLen` that's used in both places